src/Hazelcast.Net/Core/ClientVersion.cs

```diff
@@ -53,5 +57,33 @@ internal static string Version
                 return _clientVersion;
             }
         }
+
+        /// <summary>
+        /// Gets the "pure" version of this assembly.
+        /// </summary>
+        /// <remarks>
+        /// <para>Returns the pure version, ie major.minor.patch.</para>
+        /// </remarks>
+        internal static string PureVersion
+        {
+            get
+            {
+                if (_clientVersionPure != null) return _clientVersionPure;
+                var version = Version;
+                var pos0 = version.IndexOf('.');
+                var pos1 = version.IndexOf('.', pos0 + 1);
+                if (pos1 >= 0)
+                {
+                    version = version[..pos1];
+                }
+                else
+                {
+                    var pos = version.IndexOf('+');
```

Comment:
I don't get how this works with the above example semver version `5.1.2.preview-0+ab16ec43`? This has 3 dots. Do you have unit tests for this function?

Reply:
This is not a valid semver version that we use, would be `5.1.2-preview.0+ab16ec43`. Was a mistake in comments. Fixing + pushing unit tests (that never existed) and cleanup.